doc/user/inputs/blueprints.rst

```diff
@@ -404,9 +404,98 @@ material modifications
   you can define any fuel material as being made of LWR-derived TRU plus depleted uranium
   at various weight fractions. Note that this input style only adjusts the heavy metal.
 
-  .. warning:: The input processing system will try to apply the extra input parameters to all components in
-        the block, so there should typically only be one component per block that understands each input
-        parameter.
+  To enable the application of different values for the same material modification type
+  on different components within a block, the user may specify componentwise material
+  modifications. This is useful, for instance, when two pins within an assembly
+  made of the same base material have different fuel enrichments. This is done
+  using the `componentwise` attribute to the material modifications as in::
+
+        blocks:
+            fuel: &block_fuel
+                fuel1: &component_fuel_fuel1
+                    shape: Hexagon
+                    material: UZr
+                    Tinput: 600.0
+                    Thot: 600.0
+                    ip: 0.0
+                    mult: 1
+                    op: 10.0
+                fuel2: &component_fuel_fuel2
+                    shape: Hexagon
+                    material: UZr
+                    Tinput: 600.0
+                    Thot: 600.0
+                    ip: 0.0
+                    mult: 1
+                    op: 10.0
+        assemblies:
+            fuel a: &assembly_a
+                specifier: IC
+                blocks: [*block_fuel]
+                height: [1.0]
+                axial mesh points: [1]
+                xs types: [A]
+                material modifications:
+                    componentwise:
+                        fuel1:
+                            U235_wt_frac: [0.20]
+                        fuel2:
+                            Zr_wt_frac: [0.02]
+                    U235_wt_frac: [0.30]
+
+  This example would apply an enrichment of 20% to the `fuel1` component and an
```

Comment:
`fuel1` and other literals should have double-backticks around it for proper rst rendering

Reply:
Fixed.